library/core/src/panicking.rs

```diff
@@ -84,8 +84,12 @@ fn panic_bounds_check(index: usize, len: usize) -> ! {
     panic!("index out of bounds: the len is {} but the index is {}", len, index)
 }
 
+// This function is called directly by the codegen backend, and must not have
+// any extra arguments (including those synthesized by track_caller). It should
+// also not be called by regular Rust code.
```

Comment:
It's okay to call this in regular Rust code, it's just that it doesn't really make much sense.